src/components/content-tab-assistant.tsx

```diff
@@ -173,35 +214,39 @@ export function ContentTabAssistant( { selectedSite }: ContentTabAssistantProps
 	const { __ } = useI18n();
 	const lastMessage = messages.length === 0 ? undefined : messages[ messages.length - 1 ];
 
+	const [ errorMessageId, setErrorMessageId ] = useState< number | null >( null );
+	const [ isThinking, setIsThinking ] = useState( false );
+
 	useEffect( () => {
 		fetchWelcomeMessages();
 	}, [ fetchWelcomeMessages, selectedSite ] );
 
-	const handleSend = async ( messageToSend?: string ) => {
+	const handleSend = async ( messageToSend?: string, messageId?: number ) => {
 		const chatMessage = messageToSend || input;
 		if ( chatMessage.trim() ) {
-			addMessage( chatMessage, 'user', chatId );
+			const newMessageId = messageId ?? messages.length;
+			if ( messageId === undefined ) {
+				addMessage( chatMessage, 'user', chatId );
+			}
 			setInput( '' );
+			setIsThinking( true );
```

Comment:
I'm not sure if we need to keep this in the component state. In theory, `isAssistantThinking` should represent the state of waiting for the response. If it fails, it will set it to `false` ([reference](https://github.com/Automattic/studio/blob/880ff2d8301924eafdc19ecee40fc1fbf8b0b805/src/hooks/use-assistant-api.ts#L70)).

Reply:
The main reason why I had to add it is to deal with re-trying sending the message. I have now refactored my approach quite a bit so I will check if it can be removed